astropy/time/core.py

```diff
@@ -1994,6 +2016,82 @@ class TimeYearDayTime(TimeString):
                 '{year:d}:{yday:03d}'))
 
 
+class TimeFITS(TimeString):
+    """
+    FITS compliant date-time format: "[±Y]YYYY-MM-DDTHH:MM:SS.sss...(SCALE)".
+
+    ISOT with two extensions:
+    - Signed five-digit year (mostly for negative years);
+    - A possible time scale appended in parentheses.
+    """
+    name = 'fits'
+    subfmts = (
+        ('date_hms',
+         '%Y-%m-%dT%H:%M:%S',
+         '{year:04d}-{mon:02d}-{day:02d}T{hour:02d}:{min:02d}:{sec:02d}'),
+        ('date_hm',
+         '%Y-%m-%dT%H:%M',
+         '{year:04d}-{mon:02d}-{day:02d}T{hour:02d}:{min:02d}'),
+        ('date',
+         '%Y-%m-%d',
+         '{year:04d}-{mon:02d}-{day:02d}'),
+        ('longdate_hms',
+         r'(?P<year>[+-]\d{1,5})-%m-%dT%H:%M:%S',
```

Comment:
Of course putting the scale into the regex would also require not using the base `parse_string`, but maybe that's not such a bad thing since almost all of that code is not relevant in the case of a (known) pure regex.